### PR TITLE
chore(frontend): Support v1 and v2 API request simultaneously in the frontend server

### PR DESCRIPTION
**Pull request by jlyaoyuli**
Currently, the frontend server can only proxy either v1 or v2 API request to backend. The main changes of this PR as follow:

1. Separate v1 and v2 API version and version prefix in `configs.ts`
2. Add additional logic in `app.ts` for proxying v2 API request.

https://user-images.githubusercontent.com/56132941/219529919-ebab5c3f-c982-4fb4-8361-9dc36299d9cf.mov

### CODE REVIEW

File: frontend/server/app.ts [hunk 1]
```diff
@@ -92,7 +92,8 @@ export class UIServer {
 function createUIServer(options: UIConfigs) {
   const currDir = path.resolve(__dirname);
   const basePath = options.server.basePath;
-  const apiVersionPrefix = options.server.apiVersionPrefix;
+  const apiVersion1Prefix = options.server.apiVersion1Prefix;
+  const apiVersion2Prefix = options.server.apiVersion1Prefix;
```
Comment (zijianjoy) [inline frontend/server/app.ts:96]: Replace `options.server.apiVersion1Prefix` with `options.server.apiVersion2Prefix`

File: frontend/server/app.ts [hunk 2]
```diff
@@ -108,9 +109,9 @@ function createUIServer(options: UIConfigs) {
   /** Healthz */
   registerHandler(
     app.get,
-    `/${apiVersionPrefix}/healthz`,
+    `/${apiVersion1Prefix}/healthz`,
     getHealthzHandler({
-      healthzEndpoint: getHealthzEndpoint(apiServerAddress, apiVersionPrefix),
+      healthzEndpoint: getHealthzEndpoint(apiServerAddress, apiVersion1Prefix),
       healthzStats: getBuildMetadata(currDir),
     }),
```
Comment (zijianjoy) [inline frontend/server/app.ts:116]: Why are these handlers for `apiVersion1Prefix` only but not `apiVersion2Prefix`?

Comment (jlyaoyuli) [reply frontend/server/app.ts:116]: Added another one for v2

File: frontend/server/app.ts [hunk 3]
```diff
@@ -177,7 +187,7 @@ function createUIServer(options: UIConfigs) {
           /** Argo nodeId is just POD name */
           const nodeId = req.query.podname;
           const runId = req.query.runid;
-          return `/${apiVersionPrefix}/runs/${runId}/nodes/${nodeId}/log`;
```
Comment (zijianjoy) [inline frontend/server/app.ts:180]: Can you also support this in v2 as well?

File: frontend/server/app.ts [hunk 4]
```diff
@@ -229,12 +239,23 @@ function createUIServer(options: UIConfigs) {
 
   // Order matters here, since both handlers can match any proxied request with a referer,
   // and we prioritize the basepath-friendly handler
-  proxyMiddleware(app, `${basePath}/${apiVersionPrefix}`);
-  proxyMiddleware(app, `/${apiVersionPrefix}`);
```
Comment (zijianjoy) [inline frontend/server/app.ts:233]: Are these v1 only?

File: frontend/server/app.ts [hunk 5]
```diff
@@ -218,9 +228,9 @@ function createUIServer(options: UIConfigs) {
     app.use,
     [
       // Original API endpoint is /runs/{run_id}:reportMetrics, but ':reportMetrics' means a url parameter, so we don't use : here.
-      `/${apiVersionPrefix}/runs/*reportMetrics`,
-      `/${apiVersionPrefix}/workflows`,
-      `/${apiVersionPrefix}/scheduledworkflows`,
```
Comment (zijianjoy) [inline frontend/server/app.ts:223]: Shouldn't we also report 403 when these endpoints are used in v2?

File: frontend/server/app.ts (hunk 4 above)
Comment (jlyaoyuli) [reply frontend/server/app.ts:233]: Added others for v2 as well.

File: frontend/server/app.ts (hunk 5 above)
Comment (jlyaoyuli) [reply frontend/server/app.ts:223]: Great catch, `reportMetrics` is v1 only. Added both `workflow` and `scheduledworkflows` for v2.

**Comment by zijianjoy** [comment]
/lgtm
/approve

Good work adding v2 proxy!

**Pull request merged**